src/lib.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019 King's College London.
+// Created by the Software Development Team <http://soft-dev.org/>.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/// YkMir -- Metadata serialiser and deserialiser for Yorick.
+///
+/// This crate allows ykrustc to serialise the compiler's MIR structures for later deserialisation
+/// by the Yorick runtime. We refer to the data in the serialisation as "meta-data".
+///
+/// The encoder and decoder API is structured in such a way that meta-data can be streamed to/from
+/// the serialised format one item at a time. This helps to reduce memory consumption.
+///
+/// The MIR data is serialised in the msgpack format in the following form:
+///
+///  -----------
+///  version            -- The serialisation format version number.
+///                        This should be bumped every time the meta-data structure changes.
+///  md_0:             \
+///  ...⋮                - Meta-data items.
```

Comment:
I don't think we need the "⋮" character here as well as "...".

Reply:
Hah! I tried to paste a `⋮` into vim but thought it didn't work! I guess liberation mono doesn't have that glyph.